src/artemis_driver.cpp

```diff
@@ -182,7 +189,13 @@ TaskCollection ArtemisDriver<GEOM>::StepTasks() {
       const bool do_pcm = ((stage == 1) && (integrator->GetName() == "vl2"));
       TaskID gas_flx = none, dust_flx = none;
       if (do_gas) gas_flx = tl.AddTask(none, Gas::CalculateFluxes, u0.get(), do_pcm);
-      if (do_dust) dust_flx = tl.AddTask(none, Dust::CalculateFluxes, u0.get(), do_pcm);
+      if (do_dust) {
+        // update dust stopping time and dust diffusivity
+        TaskID dust_stopping_time =
+            tl.AddTask(none, Dust::UpdateDustStoppingTime<GEOM>, u0.get());
```

Comment:
This task can be removed

Reply:
I strongly suggest keeping this task and move all dust stopping time calculation to this routine because this is used in many different places. We can reduce the storage to save only the stopping time only for the first dust species, as other dust species only differ by a factor of dust size.

Reply:
Stopping time already has a function that calculates it and is used by the drag package, so at the very least any other package that uses stopping time should call that same function. I don't see the benefit of storing this, since calculating it is so cheap. But this is again one of those areas where we first implement something in a potentially non-performant way, evaluate the performance, and then potentially adjust the implementation.

Reply:
We decided previously to calculate stopping times on the fly, templating on the dust stopping model if necessary, rather than store the extra data. Because the (current) stopping models are simple, it is likely to be more performant to recalculate where necessary rather than retrieve the stored values from memory. Assuming dust stopping is proportional to dust size and hardcoding that into the code is the sort of significant complexity that we would only want to do if we measured later that dust stopping really was a bottleneck.

Reply:
Still reading through the PR... it sounds like I should expect to see these computed on the fly whenever they are needed... I agree this is what we want.  

Regardless, just a warning that by not placing these definitions in `FillDerived` (which we don't want, probably), there is some ambiguity on what these derived fields hold upon AMR remeshing (e.g., an output of this field may not contain what you expect if the output follows a remeshing event).